File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/InputsAndResultsTestPercent.jl
println("getPercent")
if isdefined(Main, :getPercent) && !isnothing(getPercent)
	error("getPercent already defined!")
end
getPercent = putPercent
println(typeof(getPercent))
println(getPercent)
if typeof(getPercent) != Float64
	error("getPercent not Float64!")
end

println("getPercentVector")
if isdefined(Main, :getPercentVector) && !isnothing(getPercentVector)
	error("getPercentVector already defined!")
end
getPercentVector = putPercentVector
println(eltype(getPercentVector))
println(getPercentVector)
if eltype(getPercentVector) != Float64
	error("getPercentVector not Float64!")
end

println("getPercentVectorAsList")
if isdefined(Main, :getPercentVectorAsList) && !isnothing(getPercentVectorAsList)
	error("getPercentVectorAsList already defined!")
end
getPercentVectorAsList = putPercentVectorAsList
println(eltype(getPercentVectorAsList))
println(getPercentVectorAsList)
if eltype(getPercentVectorAsList) != Float64
	error("getPercentVectorAsList not Float64!")
end

println("getPercentMatrix")
if isdefined(Main, :getPercentMatrix) && !isnothing(getPercentMatrix)
	error("getPercentMatrix already defined!")
end
getPercentMatrix = putPercentMatrix
println(eltype(getPercentMatrix))
println(getPercentMatrix)
if eltype(getPercentMatrix) != Float64
	error("getPercentMatrix not Float64!")
end

println("getPercentMatrixAsList")
if isdefined(Main, :getPercentMatrixAsList) && !isnothing(getPercentMatrixAsList)
	error("getPercentMatrixAsList already defined!")
end
getPercentMatrixAsList = putPercentMatrixAsList
println(eltype(getPercentMatrixAsList))
println(getPercentMatrixAsList)
if eltype(getPercentMatrixAsList) != Float64
	error("getPercentMatrixAsList not Float64!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestPercent.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getPercent")
if(getPercent ~= nil) then
	error("getPercent already defined!")
end
getPercent = putPercent
getPercentType = type(getPercent)
print(getPercentType)
print(getPercent)
if(getPercentType ~= "number") then
	error("getPercent not Double!")
end

print("getPercentVector")
if(getPercentVector ~= nil) then
	error("getPercentVector already defined!")
end
getPercentVector = putPercentVector
getPercentVectorType = type(getPercentVector[1])
print(getPercentVectorType)
print(arraysToString(getPercentVector))
if(getPercentVectorType ~= "number") then
	error("getPercentVector not Double!")
end

print("getPercentVectorAsList")
if(getPercentVectorAsList ~= nil) then
	error("getPercentVectorAsList already defined!")
end
getPercentVectorAsList = putPercentVectorAsList
getPercentVectorAsListType = type(getPercentVectorAsList[1])
print(getPercentVectorAsListType)
print(arraysToString(getPercentVectorAsList))
if(getPercentVectorAsListType ~= "number") then
	error("getPercentVectorAsList not Double!")
end

print("getPercentMatrix")
if(getPercentMatrix ~= nil) then
	error("getPercentMatrix already defined!")
end
getPercentMatrix = putPercentMatrix
getPercentMatrixType = type(getPercentMatrix[1][1])
print(getPercentMatrixType)
print(arraysToString(getPercentMatrix))
if(getPercentMatrixType ~= "number") then
	error("getPercentMatrix not Double!")
end

print("getPercentMatrixAsList")
if(getPercentMatrixAsList ~= nil) then
	error("getPercentMatrixAsList already defined!")
end
getPercentMatrixAsList = putPercentMatrixAsList
getPercentMatrixAsListType = type(getPercentMatrixAsList[1][1])
print(getPercentMatrixAsListType)
print(arraysToString(getPercentMatrixAsList))
if(getPercentMatrixAsListType ~= "number") then
	error("getPercentMatrixAsList not Double!")
end
